code/modules/mob/living/simple_animal/abnormality/waw/snow_whites_apple.dm

```diff
@@ -256,6 +272,16 @@
 	if(!ishuman(L))
 		L.adjustStaminaLoss(10, TRUE, TRUE) //nonhumans are hindered more
 
+/obj/structure/alien/weeds/F0442/proc/get_abno(mob/living/simple_animal/hostile/abnormality/snow_whites_apple/SWA)
```

Comment:
Why do we even need it?

Reply:
the issue is setting the variable BEFORE the vines are created is, as far as I can tell, impossible. I could make the vine check on Initialize for every datum to find snow, but it's a lot more clean to have it be attributed to a proc instead. If the proc is (somehow?) happening AFTER its death, it'll delete itself anyway as a failsafe.

In other words it's mostly for convenience.